execution/src/execution.rs

```diff
@@ -345,6 +346,108 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
         }
         Ok(logs)
     }
+
+    pub async fn get_fee_history(
+        &self,
+        block_count: u64,
+        last_block: u64,
+        reward_percentiles: &[f64],
+        payloads: &BTreeMap<u64, ExecutionPayload>,
+    ) -> Result<Option<FeeHistory>> {
+        let helios_latest_block = payloads.last_key_value();
+        let helios_latest_block = *helios_latest_block
+            .ok_or(ExecutionError::EmptyExecutionPayload())?
+            .0;
+        let helios_oldest_block = payloads.first_key_value();
+        let helios_oldest_block = *helios_oldest_block
+            .ok_or(ExecutionError::EmptyExecutionPayload())?
+            .0;
+        let mut block_count = block_count;
+        let mut request_latest_block = last_block;
+        //(I would also allow BlockNotFoundError to accept an optional BlockTag so we can construct an error without knowing the actual block)
+        debug!(
+            "get_fee_history parameters:
+               helios_latest_block {:?}, helios_oldest_block {:?},
+               block_count {:?}, request_latest_block {:?}, reward_percentiles {:?}",
+            helios_latest_block,
+            helios_oldest_block,
+            block_count,
+            request_latest_block,
+            reward_percentiles
+        );
+
+        //case: requested block is more recent than helios' newest block
+        if request_latest_block > helios_latest_block {
+            request_latest_block = helios_latest_block;
+        }
+
+        //Can't prove anything in the request range
+        if request_latest_block < helios_oldest_block {
+            return Err(ExecutionError::InvalidBlockRange(
+                request_latest_block,
+                helios_latest_block,
+            )
+            .into());
+        }
+
+        let request_oldest_block = request_latest_block - block_count;
+        //case: request oldest block is further out than what helios' is aware of
+        if request_oldest_block < helios_oldest_block {
+            block_count = request_latest_block - helios_oldest_block + 1;
+        }
+
+        let fee_history = self
+            .rpc
+            .get_fee_history(block_count, request_latest_block, reward_percentiles)
```

Comment:
Do we actually need to call the rpc here? As we discussed, I think we were planning on omitting the reward percentiles since we cannot verify them. In that case, the remaining fields can all be fully computed (as you do here). Rather than computing and checking for correctness, we can just use the computed values.